### PR TITLE
gives the server control over the clients login options

### LINKED ISSUE #2076: Do not disable local account creation when an IdP is configured
> https://github.com/Open-MSS/MSS/blob/f6b3ad110dbfaa12d439cadea6476dcd54d9328d/mslib/msui/mscolab.py#L242-L245
> Shouldn't it still be possible to register a new non-IdP user on the MSColab instance, even if an IdP is configured? I feel like disabling "local" account registration should be a different config option (that would need to be enforced server-side).

_Originally posted by @matrss in https://github.com/Open-MSS/MSS/pull/2069#discussion_r1384874147_
            

> also in the workflow of profile/usercreation by the IDP is not completed. The new profile is not activated, it would need a validation (recover password attempt currently) first. There some changes we have to think about after we merged this in.

_Originally posted by @ReimarBauer in https://github.com/Open-MSS/MSS/pull/2069#discussion_r1384930546_

> We have to verify too that a user with a "None" Password because of IDP can't fetch one and update that 'None' over password recovery.  Some of these parts need a coverage by tests.

_Originally posted by @ReimarBauer in https://github.com/Open-MSS/MSS/pull/2069#discussion_r1384946717_

### PR DESCRIPTION
**Pull request by ReimarBauer**
**Purpose of PR?**:

Fixes #2076

Dependent of the server configuration a user login to a local database can be archieved.

### CODE REVIEW

**Comment by matrss** [comment]
This should be enforced on the server side as well, i.e. disable all endpoints related to local user registration and login.

**Comment by ReimarBauer** [comment]
There is more to do also after this is done https://github.com/Open-MSS/MSS/issues/2078

I added already on some places ToDo's. I think we do this step by step and at the end another one. Some parts are not identified which needs changes too.

File: mslib/mscolab/server.py
Comment (matrss) [inline mslib/mscolab/server.py:246]: We should decide if keys in this json should be uppercase or lowercase, not mix them. I'd prefer lowercase.

Comment (nilupulmanodya) [reply mslib/mscolab/server.py:246]: Yeah.. Thank you for figguring out that point. I believe that the `USE_SAML2` configuration pertains to a previous implementation and is not directly relevant to the current issue at hand. Consequently, it may be prudent to consider opening a separate issue, as the matter at hand is distinct and not associated with the present concern. This step is suggested to enhance clarity and maintain a focused approach to problem resolution.

**Comment by nilupulmanodya** [comment]
We have TODOs. Given that the user login to a local database is achievable depending on the server configuration, I am granting approval for this pull request.

**Review by nilupulmanodya — APPROVED** [review]

File: mslib/mscolab/server.py
Comment (matrss) [reply mslib/mscolab/server.py:246]: If we decide on lowercase then yes, this does not block this PR and should be an issue instead. I just wanted to bring it up, in case someone disagreed with me preferring lowercase.

**Review by matrss — APPROVED** [review]

**Pull request merged**
